torch/_inductor/select_algorithm.py

```diff
@@ -1198,22 +1346,166 @@ def generate(  # type: ignore[override]
                         f"num_buffers_warp_spec={num_buffers_warp_spec}",
                     ]
                 )
-
             extra = "-".join(extra_parts) + "-"
-            mod = PyCodeCache.load(code, extra)
```

Comment:
why do we need to do this ?

Reply:
this function is make_extra is just computing extra and returning it.
unrelated but not sure if this what you are asking about. 
mod also is not cached so its moved outside the caching lookup block. (it has its own cache)